src/conv/problem_description.cpp

```diff
@@ -64,7 +64,7 @@ void ProblemDescription::HeuristicUpdateLayouts()
 {
     const std::string labels = tensor_layout_get_default(in_layout.size());
 
-    static const std::vector<std::string> supported_layouts = {"NCHW", "NHWC", "NCDHW"};
+    static const std::vector<std::string> supported_layouts = {"NCHW", "NHWC", "CHWN", "NCDHW"};
```

Comment:
In PR #1523 , convolution kernel with NCHWc(input tensor) x CHWNc (weight tensor) = NCHWc (output tensor) are implemented, that's why I include CHWN layout into supported layouts. I'm not sure a separate PR is possible or not, since we only use CHWNc layout as weight tensor.

Reply:
@aska-0096 
> ...we only use CHWNc layout as weight tensor.

Sorry but I do not understand still... There is no "N" in the weights tensor.

IIRC we derive weights layout from input & output layouts, implicitly. For example, for NCHW input and output, weights is of KCYX layout. Ditto NHWC convolutions. We do not require the user to explicitly specify weights layout for both NCHW and NHWC.

Why this is necessary for NCHWc layout?

Reply:
Hi @atamazov.
`CHWNc` means `CYXKc`  in weight tensor.

> IIRC we derive weights layout from input & output layouts, implicitly. For example, for NCHW input and output, weights is of KCYX layout.

Yes, it is, but not necessary. The convolution is all about multiply two tensors and get another one, in forward convolution, we always want input tensor and output tensor have same layout while weight tensor layout is arbitrary, because as you can imagine, tensor layout is just the order of data lied in computer memory space. The reason we introduce `NCHWc x CYXKc = NCHWc` (where weight tensor is different from input and output) is just because it's faster in some cases and we support it in MISA(implicit gemm generator).

> We do not require the user to explicitly specify weights layout for both NCHW and NHWC


Actually we need, but we did it implicitly in based on two MIOpen logics:

1. Default tensor layout is NCHW.
2. Input tensor, weight tensor and output tensor have same tensor layout.

These legacy only native support NCHW, which means, lack of flexibility.
NHWC support was did by carlushuang, and he put much effort on transform NHWC tensor to NCHW tensor.

> So the question is: Why setting the layout of weights is necessary for NCHWc?


It's natural logic in convolution if we want to support different tensor layouts. As I talked above, the biggest reason is in the forward convolution:
**weight tensor could be in arbitrary layout.**

Reply:
@aska-0096 Thanks, agreed.